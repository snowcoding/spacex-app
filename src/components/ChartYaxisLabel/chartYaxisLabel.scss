@import '../../globals.scss';

.y-axis-label{
    display:flex;
    align-items: center;

    .fas.fa-info-circle{
        font-size: 1.4em;
        margin-left: 12px;
        color:$mui-primary;
        padding-top:5px;
    }
}