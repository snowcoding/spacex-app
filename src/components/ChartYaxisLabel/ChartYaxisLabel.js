import React from 'react'
import Popover from '@material-ui/core/Popover'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'
import { colors } from '@material-ui/core'
import './chartYaxisLabel.scss'

const useStyles = makeStyles(theme => ({
  popOverText: {
    padding: theme.spacing(2),
    maxWidth: 300,
  },
  popover: {
    pointerEvents: 'none',
  },
  paper: {
    padding: theme.spacing(1),
  },
  helperText: {
    color: colors.blue,
  },
}))

export default function ChartYaxisLabel(props) {
  const classes = useStyles()
  const [anchorEl, setAnchorEl] = React.useState(null)
  const { yAxisLabel } = props

  function handlePopoverOpen(event) {
    setAnchorEl(event.currentTarget)
  }

  function handlePopoverClose() {
    setAnchorEl(null)
  }

  const open = Boolean(anchorEl)

  return (
    <div>
      <a id={yAxisLabel.anchor}>
      <div className='y-axis-label'>
        <Typography variant='h5' color='primary' className='title'>
          {yAxisLabel.title}
        </Typography>
        <i
          className='fas fa-info-circle'
          aria-owns={open ? 'mouse-over-popover' : undefined}
          aria-haspopup='true'
          onMouseEnter={handlePopoverOpen}
          onMouseLeave={handlePopoverClose}
        />
      </div></a>
      <Popover
        id='mouse-over-popover'
        className={classes.popover}
        classes={{
          paper: classes.paper,
        }}
        open={open}
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'right',
        }}
        transformOrigin={{
          vertical: 'top',
          horizontal: 'left',
        }}
        onClose={handlePopoverClose}
        disableRestoreFocus
      >
        <Typography className={classes.popOverText}>
          {yAxisLabel.popover}
          {/* Come up with a way to link to the wikipedia page */}
          {/* {yAxisLabel.link ? yAxisLabel.link() : null} */}
        </Typography>
      </Popover>
    </div>
  )
}
