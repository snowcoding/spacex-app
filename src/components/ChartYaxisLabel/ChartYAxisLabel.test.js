import React from 'react'
import ReactDOM from 'react-dom'
import ChartYaxisLabel from './ChartYaxisLabel'
import renderer from 'react-test-renderer'

it('snapshot test Y axis Label of Chart', () => {
  const yAxisLabel = { title: 'title', popover: 'popever', link: 'link' }
  const tree = renderer
    .create(<ChartYaxisLabel yAxisLabel={yAxisLabel} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders without crashing', () => {
  const div = document.createElement('div')
  const yAxisLabel = { title: 'title', popover: 'popever', link: 'link' }
  ReactDOM.render(<ChartYaxisLabel yAxisLabel={yAxisLabel} />, div)
  ReactDOM.unmountComponentAtNode(div)
})
