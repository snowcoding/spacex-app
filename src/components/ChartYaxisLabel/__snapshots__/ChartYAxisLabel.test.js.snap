// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Y axis Label of Chart 1`] = `
<div>
  <h5
    aria-haspopup="true"
    className="MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    title
  </h5>
</div>
`;
